### PR TITLE
[E1031] Ensure determine-reboot-cause start after platform fully initialized

### PR DESCRIPTION
**Pull request by lizhijianrd**
Echo #18507 and fix comments.

#### Why I did it

When determine-reboot-cause is run early, there is very chance it will fail on call to Chassis(), which calls Thermal(), because it checks hwmon that was not yet initialized.

##### Work item tracking
- Microsoft ADO **(number only)**: 27456392

#### How I did it

Update platform-modules-haliburton.service file. Let determine-reboot-cause.service start after platform fully initialized.

#### How to verify it

Built dev image based on 202305 branch.
1. Manually reboot multiple times to confirm determine-reboot-cause.service always start currectly.
2. Run nightly test and confirmed no regression introduced.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305
- [x] 202311

#### Tested branch (Please provide the tested image version)

- [x] 202305 
- [ ]

### CODE REVIEW

File: device/celestica/x86_64-cel_e1031-r0/sonic_platform/thermal.py [hunk 1]
```diff
@@ -109,12 +109,16 @@ def __init__(self, thermal_index, airflow):
...
         self.postion = self._thermal_info["postion"]
-        self.minimum_thermal = self.get_temperature()
-        self.maximum_thermal = self.get_temperature()
+        self.minimum_thermal = None
+        self.maximum_thermal = None
 
     def _get_hwmon_path(self):
         hwmon_path = os.path.join(
             I2C_ADAPTER_PATH, self._thermal_info["i2c_path"])
+        if self._thermal_info["i2c_path"] == "i2c-11/11-001a/hwmon":
```
Comment (prgeor) [inline device/celestica/x86_64-cel_e1031-r0/sonic_platform/thermal.py:118]: @lizhijianrd exact path depends upon the hwmon enumerating the devices. We don't need this hard coding here. Ensuring the platform driver gets installed before reboot cause determination process runs is sufficient to fix the issue

Comment (lizhijianrd) [reply device/celestica/x86_64-cel_e1031-r0/sonic_platform/thermal.py]: Thanks for pointing this out. Updated.

**Comment by lizhijianrd** [comment]
Hi @yxieca , can you please help to review and merge this PR?

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@lguohan please merge/review

**Pull request merged**